Build failure: wireplumber-0.4.17 | pwucontrol-0.4.8

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

{ pkgs, ... }:{

  environment.systemPackages = with pkgs; [
    pwvucontrol
  ];
}

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

500 Internal Server Error - I get this instead of answers from hydra website

### Relevant log output

```console
error: builder for '/nix/store/fkczglbi9k3r25hm7rv21g7s9jqvmid1-wireplumber-0.4.17.drv' failed with exit code 1;
       last 25 log lines:
       > gcc -Ilib/wp/libwireplumber-0.4.so.0.4.17.p -Ilib/wp -I../lib/wp -Ilib -I../lib -I/nix/store/bry9cijf53mjxhfcwv77ps5y2f9zak8k-glib-2.82.4-dev/include -I/nix/store/bry9cijf53mjxhfcwv77ps5y2f9zak8k-glib-2.82.4-dev/include/glib-2.0 -I/nix/store/7ly9p55j21sr0rl93lb6vpi1k1sbb3x3-glib-2.82.4/lib/glib-2.0/include -I/nix/store/aw3g4x0ski0004j27kc51vf2vh7zawck-pipewire-1.4.0-dev/include/pipewire-0.3 -I/nix/store/aw3g4x0ski0004j27kc51vf2vh7zawck-pipewire-1.4.0-dev/include/spa-0.2 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_62 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_62 '-DGETTEXT_PACKAGE="wireplumber"' -fvisibility=hidden -Wsuggest-attribute=format -Wsign-compare -Wpointer-arith -Wpointer-sign -Wformat -Wformat-security -Wimplicit-fallthrough -Wmissing-braces -Wtype-limits -Wvariadic-macros -Wno-missing-field-initializers -Wno-unused-parameter -Wno-pedantic -Wold-style-declaration -Wunused-result -fPIC -D_REENTRANT -pthread -D_GNU_SOURCE -DG_LOG_USE_STRUCTURED '-DWIREPLUMBER_DEFAULT_MODULE_DIR="/nix/store/v5l8xydqyn8jg5s7w7k1k7702id0jx5q-wireplumber-0.4.17/lib/wireplumber-0.4"' '-DWIREPLUMBER_DEFAULT_CONFIG_DIR="/etc/wireplumber"' '-DWIREPLUMBER_DEFAULT_DATA_DIR="/nix/store/v5l8xydqyn8jg5s7w7k1k7702id0jx5q-wireplumber-0.4.17/share/wireplumber"' '-DLOCALE_DIR="/nix/store/v5l8xydqyn8jg5s7w7k1k7702id0jx5q-wireplumber-0.4.17/share/locale"' -DBUILDING_WP -MD -MQ lib/wp/libwireplumber-0.4.so.0.4.17.p/node.c.o -MF lib/wp/libwireplumber-0.4.so.0.4.17.p/node.c.o.d -o lib/wp/libwireplumber-0.4.so.0.4.17.p/node.c.o -c ../lib/wp/node.c
       > ../lib/wp/node.c: In function 'wp_node_send_command':
       > ../lib/wp/node.c:634:25: error: passing argument 1 of 'pw_node_send_command' from incompatible pointer type [-Wincompatible-pointer-types]
       >   634 |   pw_node_send_command (wp_proxy_get_pw_proxy (WP_PROXY (self)), &cmd);
       >       |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       >       |                         |
       >       |                         struct pw_proxy *
       > In file included from /nix/store/aw3g4x0ski0004j27kc51vf2vh7zawck-pipewire-1.4.0-dev/include/pipewire-0.3/pipewire/pipewire.h:30,
       >                  from ../lib/wp/private/pipewire-object-mixin.h:14,
       >                  from ../lib/wp/node.c:16:
       > /nix/store/aw3g4x0ski0004j27kc51vf2vh7zawck-pipewire-1.4.0-dev/include/pipewire-0.3/pipewire/node.h:227:59: note: expected 'struct pw_node *' but argument is of type 'struct pw_proxy *'
       >   227 | PW_API_NODE_IMPL int pw_node_send_command(struct pw_node *object, const struct spa_command *command)
       >       |                                           ~~~~~~~~~~~~~~~~^~~~~~
       > [82/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/module.c.o
       > [83/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/plugin.c.o
       > [84/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/object.c.o
       > [85/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/metadata.c.o
       > [86/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/port.c.o
       > [87/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/proxy-interfaces.c.o
       > [88/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/proxy.c.o
       > [89/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/log.c.o
       > [90/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/object-interest.c.o
       > [91/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/properties.c.o
       > [92/207] Compiling C object lib/wp/libwireplumber-0.4.so.0.4.17.p/object-manager.c.o
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/fkczglbi9k3r25hm7rv21g7s9jqvmid1-wireplumber-0.4.17.drv'.
error: 1 dependencies of derivation '/nix/store/hk9xsyjhw3k804p9nmd5c6xdj3pqri8h-pwvucontrol-0.4.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3mwcwbsa47iyadfdl2qd3lpaic40hg5m-system-path.drv' failed to build
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.6, NixOS, 25.05 (Warbler), 25.05.20250309.e3e32b6`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/nix/store/g4ppspdl4fy7hnp4jgjl4ll03v7i08w3-source`

### Notify maintainers

@figsoda 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pw/pwvucontrol/package.nix
+++ b/pkgs/by-name/pw/pwvucontrol/package.nix
@@ -3,6 +3,7 @@
   stdenv,
   fetchFromGitHub,
   fetchFromGitLab,
+  fetchpatch,
   cargo,
   desktop-file-utils,
   meson,
@@ -31,6 +32,13 @@ let
       tag = version;
       hash = "sha256-vhpQT67+849WV1SFthQdUeFnYe/okudTQJoL3y+wXwI=";
     };
+
+    patches = [
+      (fetchpatch {
+        url = "https://gitlab.freedesktop.org/pipewire/wireplumber/-/commit/f4f495ee212c46611303dec9cd18996830d7f721.patch";
+        hash = "sha256-dxVlXFGyNvWKZBrZniFatPPnK+38pFGig7LGAsc6Ydc=";
+      })
+    ];
   });
 in
 stdenv.mkDerivation (finalAttrs: {

